ui/src/screens/Jobs/JobsScreen.tsx

```diff
@@ -66,19 +74,27 @@ const JobsScreen: React.FC = function () {
       <Search placeholder="Search job postings" onChange={onSearchChange} />
       <CategorySelector category="jobs" />
       <PostsContainer>
-        {state.jobs.map((post: PostInterface) => (
-          <Post category="jobs" active={post.active} key={post.id} data={post} />
-        ))}
-        <PostLoader hasMore={state.more} onLoadMore={handleLoadMore} />
+        {state.fetched ? (
+          <>
+            {state.jobs.map((post: PostInterface) => (
+              <PostBlock key={post.id}>
```

Comment:
I had to shift this PostBlock styled component out of `PostContainer` as for some reason the app was not rendering the `PostBlock` in `PostContainer` when within a React fragment

Reply:
Ah I see why. Adding the `<>` means that we are no longer passing an array of `<Post>`s into PostContainer where they are individually wrapped by `PostBlock`. Instead it reads the fragment wrapper as a single element and wraps a single `PostBlock` around it.
Your solution brings `PostBlock` out and wraps the posts individually first before sending them into PostContainer, **which is fine**.

That means in `PostContainer`, `React.Children.map` does nothing at all (it takes takes and array of posts, and returns the same array of posts).
It can simply be this now:
```
...
return <Container>{children}</Container>
```

**For argument's sake**, if we wanted to keep `PostBlock` in PostContainer and manipulate Jobs/TalentsScreen in such a way to make it work, it could look like this:
```
{state.fetched ? (
  [
    ...state.jobs.map((post: PostInterface) => (
      <Post category="jobs" active={post.active} key={post.id} data={post} />
    )),
  ].concat(<PostLoader hasMore={state.more} onLoadMore={handleLoadMore} />)
) : (
  <PostSpinner />
)}
```
Notice that I need to get rid of the fragment wrapper `<>`, and pass a single array of elements into PostContainer.

In order to get the mapped jobs and the PostLoader into the same array, i have to spread the mapped jobs into an array and concat the PostLoader. I'd consider this convoluted though so don't do this. Hope it helps in clarifying your understanding.

Will approve this PR but please do these changes:

- [ ] Update `PostContainer` to not use `React.Children.map`
- [ ] Update Jobs/TalentsScreen: remove `key={....id}` from `<Post ...>`, since PostBlock already has `key`.

Reply:
Thanks for the detailed explanation. The official React docs don't really cover stuff like this.

Agreed on the second example of using a spread operator with concatenation to be too convoluted.